### PR TITLE
Remove `ancilla` argument in `iterative_qpe`

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**
as title
**Description of the Change:**
Here are the occurrences I noticed to move:
 - src: `qml.ops.functions.iterative_qpe.py`
 - test of the src mentioned above
 - 
**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

**Related GitHub Issues:**
[sc-77483]

### CODE REVIEW

**Comment by JerryChen97** [comment]
@lillian542 @mudit2812

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -9,16 +9,12 @@
...
-      - CX (controlled-X)
-      - Multi-controlled X gates
-      - Grover operators
-  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
+
+  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) An `apply_operation` helper function featuring:
+
+  * Two density matrix contraction methods using `einsum` and `tensordot`
+
+  * Optimized handling of special cases including: Diagonal operators, Identity operators, CX (controlled-X), Multi-controlled X gates, Grover operators
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:17]: This changelog entry might've been a merge artifact, but it's likely a duplicate so should be resolved.

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -28,6 +24,9 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The ``'ancilla'`` argument for :func:`~pennylane.iterative_qpe` has been removed. Instead, use the ``'aux_wire'``.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:27]:
```suggestion
* The ``'ancilla'`` argument for :func:`~pennylane.iterative_qpe` has been removed. Instead, use the ``'aux_wire'`` argument.
```

File: pennylane/ops/functions/iterative_qpe.py [hunk 3]
```diff
@@ -15,30 +15,22 @@
...
 
-from warnings import warn
 
 import numpy as np
 
 import pennylane as qml
 
 
-def iterative_qpe(base, aux_wire="unset", iters="unset", ancilla="unset"):
+def iterative_qpe(base, aux_wire="unset", iters="unset"):
```
Comment (mudit2812) [inline pennylane/ops/functions/iterative_qpe.py:24]: These are meant to be required, positional arguments, and were converted into "required" keyword arguments as part of the deprecation.

```suggestion
def iterative_qpe(base, aux_wire, iters):
```

File: pennylane/ops/functions/iterative_qpe.py [hunk 4]
```diff
@@ -84,7 +76,7 @@ def circuit():
                                                                                                                       ╚═══════╡ ╰Sample[MCM]
     """
     missing = []
-    if aux_wire == "unset" and ancilla == "unset":
+    if aux_wire == "unset":
         missing.append("'aux_wire'")
     if iters == "unset":
         missing.append("'iters'")
```
Comment (mudit2812) [inline pennylane/ops/functions/iterative_qpe.py:82]: You can remove this whole section once the arguments are changed to positional arguments.

**Review by mudit2812 — CHANGES_REQUESTED** [review]
Looks good. I've left a couple of blocking comments. Happy to approve once resolved.

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (JerryChen97) [reply doc/releases/changelog-dev.md:17]: @mudit2812 Yeah this is from docs fixing conversation from another pr [here](https://github.com/PennyLaneAI/pennylane/pull/6503). This corrected docs also appear in my other removal pr's. Anyway the orignal docstrings are not rendered correctly and will be fixed; I can also create another PR separately to fix it. What do you think?

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -20,16 +20,12 @@
...
-      - CX (controlled-X)
-      - Multi-controlled X gates
-      - Grover operators
-  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
+
+  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379) An `apply_operation` helper function featuring:
+
+  * Two density matrix contraction methods using `einsum` and `tensordot`
+
+  * Optimized handling of special cases including: Diagonal operators, Identity operators, CX (controlled-X), Multi-controlled X gates, Grover operators
```
Comment (JerryChen97) [inline doc/releases/changelog-dev.md:28]:
```suggestion
  - A new API for mixed-state operations
  - An `apply_operation` helper function featuring:
    - Two density matrix contraction methods using `einsum` and `tensordot`
    - Optimized handling of special cases including:
      - Diagonal operators
      - Identity operators 
      - CX (controlled-X)
      - Multi-controlled X gates
      - Grover operators
  [(#6379)](https://github.com/PennyLaneAI/pennylane/pull/6379)
```

File: doc/releases/changelog-dev.md (hunk 1 above)
Comment (JerryChen97) [reply doc/releases/changelog-dev.md]: Reverted

File: doc/releases/changelog-dev.md [hunk 6]
```diff
@@ -39,6 +39,9 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The ``'ancilla'`` argument for :func:`~pennylane.iterative_qpe` has been removed. Instead, use the ``'aux_wire'`` argument.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:42]:
```suggestion
* The `'ancilla'` argument for `qml.iterative_qpe` has been removed. Instead, use the `'aux_wire'` argument.
```

File: tests/ops/functions/test_iterative_qpe.py [hunk 7]
```diff
@@ -23,33 +23,19 @@
...
     @pytest.mark.parametrize(
         "args, n_missing, missing_args",
         [
             ({"aux_wire": 1}, 1, "'iters'"),
-            ({"ancilla": 1}, 1, "'iters'"),
             ({"iters": 1}, 1, "'aux_wire'"),
             ({}, 2, "'aux_wire' and 'iters'"),
         ],
     )
     def test_args_not_provided(self, args, n_missing, missing_args):
```
Comment (mudit2812) [inline tests/ops/functions/test_iterative_qpe.py:34]: You can remove this whole test. It was added to test that whole section in `iterative_qpe` for checking for missing args.

**Review by mudit2812 — COMMENTED** [review]
Final comments, otherwise ready to approve :)

**Review by mudit2812 — APPROVED** [review]

**Review by lillian542 — APPROVED** [review]
🎉

**Comment by JerryChen97** [comment]
closed due to the bundled PR already in

**Pull request closed without merging**